torch/distributed/rpc/api.py

```diff
@@ -454,9 +456,32 @@ def rpc_sync(to, func, args=None, kwargs=None):
         >>> import torch.distributed.rpc as rpc
         >>> rpc.init_rpc("worker1", rank=1, world_size=2)
         >>> rpc.shutdown()
+
+        If invoking an annotated TorchScript function, then run the following
+        code in two different processes:
+
+        >>> # On worker 0:
+        >>> @torch.jit.script
+        >>> def my_script_add(t1, t2):
+        >>>    return torch.add(t1, t2)
+        >>> import torch.distributed.rpc as rpc
+        >>> rpc.init_rpc("worker0", rank=0, world_size=2)
+        >>> ret = rpc.rpc_sync("worker1", my_script_add, args=(torch.ones(2), 3))
+        >>> rpc.shutdown()
+
+        >>> # On worker 1:
+        >>> import torch.distributed.rpc as rpc
+        >>> rpc.init_rpc("worker1", rank=1, world_size=2)
+        >>> rpc.shutdown()
+
     """
-    fut = _invoke_rpc(to, func, RPCExecMode.SYNC, args, kwargs)
-    return fut.wait()
+    # If invoking an annotated TorchScript function,
+    # call the internal API _rpc_sync_torchscript()
+    if isinstance(func, torch.jit.ScriptFunction):
+        return _rpc_sync_torchscript(to, _qualified_name(func), args, kwargs)
```

Comment:
Can we put this into `def _invoke_rpc` ?

Also, `def _invoke_rpc` could be renamed to something similar to "dispatch_rpc" to make the functionality of it clearer.
As it dispatches a Python rpc call to one of the 3 code paths.

Reply:
for autograd profiler, I will address that in follow up diff with more tests, as this diff is already a little bit big:)

Reply:
tracked in PR #32375